scheduler/services/visibility/calculator.py

```diff
@@ -153,13 +152,12 @@ class VisibilityCalculator(metaclass=Singleton):
                                                           Semester(2019, SemesterHalf('A')),
                                                           Semester(2019, SemesterHalf('B'))])
 
-    def __init__(self, with_redis: bool = True):
+    def __init__(self):
         self.vis_table: Optional[TargetVisibilityTable] = None
-        self.with_redis = with_redis
 
-    async def calculate(self) -> None:
-        if self.with_redis:
-            all_semesters_vis_table = {}
+    async def calculate(self, with_redis: bool = True, programs_ids: Optional[bytes] = None) -> None:
```

Comment:
Just curious, why change it from a class variable to a parameter, is there a case when we have the calculator doing both with and without redis in run time?